### PR TITLE
Replace `cli-clipboard` with `arboard`

### LINKED ISSUE #2028: [Bug]: Copy to clipboard unsuccessful
### What did you expect to happen?

Pressing `ctrl-y` in the search view should copy the selected line to the clipboard

### What happened?

Pressing `ctrl-y` closes atuin but the clipboard content is not set 

### Atuin doctor output

```yaml
atuin:
  version: 18.2.0
  sync:
    cloud: false
    records: true
    auto_sync: true
    last_sync: 2024-05-17 11:10:19.493442155 +00:00:00
shell:
  name: fish
  default: unknown
  plugins:
  - atuin
system:
  os: Fedora Linux
  arch: x86_64
  version: '39'
  disks:
  - name: /dev/mapper/luks-a71d0523-7b8b-4cc2-9d60-a73e65041027
    filesystem: btrfs
  - name: /dev/mapper/luks-a71d0523-7b8b-4cc2-9d60-a73e65041027
    filesystem: btrfs
  - name: /dev/nvme0n1p2
    filesystem: ext4
  - name: /dev/nvme0n1p1
    filesystem: vfat
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by LecrisUT**
Unfortunately I don't think this is covered by the tests, so I am not sure how correct it is. I've mostly followed the example [here](https://github.com/1Password/arboard/blob/master/examples/set_html.rs)

## Checks
- [x] I am happy for maintainers to push small adjustments to this PR, to speed up the review cycle
- [x] I have checked that there are no existing pull requests for the same thing

---

closes #2028

@ovv Could you give it a try? the current fix here seems to work on my end, but I don't think it's reliable

### CODE REVIEW

**Comment by LecrisUT** [comment]
@ovv, seems that `arboard` has a similar issue?

> Note that in my tests the wayland backend did not keep the clipboard contents after the process exited. (Although neither did the X11 backend on my Wayland setup).

Maybe the same fix you had in #2028 works here?

File: crates/atuin/Cargo.toml [hunk 1]
```diff
@@ -87,8 +87,11 @@ serde_yaml = "0.9.32"
 sysinfo = "0.30.7"
 regex="1.10.4"
 
-[target.'cfg(any(target_os = "windows", target_os = "macos", target_os = "linux"))'.dependencies]
-cli-clipboard = { version = "0.4.0", optional = true }
+[target.'cfg(any(target_os = "windows", target_os = "macos"))'.dependencies]
+arboard = { version = "3.4", optional = true }
+
+[target.'cfg(target_os = "linux")'.dependencies]
+arboard = { version = "3.4", optional = true, features = ["wayland-data-control"] }
```
Comment (LecrisUT) [inline crates/atuin/Cargo.toml:94]: Not sure about this one, but would want to have at least an option to enable the native wayland support.

File: crates/atuin/src/command/client/search/interactive.rs [hunk 2]
```diff
@@ -1174,7 +1174,10 @@ pub async fn history(
     any(target_os = "windows", target_os = "macos", target_os = "linux")
 ))]
 fn set_clipboard(s: String) {
-    cli_clipboard::set_contents(s).unwrap();
+    let mut ctx = arboard::Clipboard::new().unwrap();
+    ctx.set_text(s.clone()).unwrap();
+    // Use the clipboard context to make sure it is saved
+    ctx.get_text().unwrap();
```
Comment (LecrisUT) [inline crates/atuin/src/command/client/search/interactive.rs:1180]: The `get_text()` seems to make sure that the clipboard content is saved, solving #2028. But I think it should be in a short loop to check the contents are equal. Any preference on what loop to do here?

**Comment by LecrisUT** [comment]
Note: I seem to encounter a bug where I can't copy something from a window (maybe XWayland), but if I try on another window (maybe native wayland) to copy it recovers. I can't seem to reliably reproduce, but it reminds me of a bug that used to happen in KeepassXC: https://github.com/keepassxreboot/keepassxc/issues/8492#issuecomment-1257334063

**Review by ellie — APPROVED** [review]
I'm not sure happy with the clipboard state on linux, but this seems fine for now.

If we face any more weird edge cases, I'd rather disable the clipboard feature by default on Linux than either have unreliable behaviour or a sleep introduced.

**Review by ellie — DISMISSED** [review]

**Review by ellie — APPROVED** [review]

**Comment by diplodocuslongus** [comment]
> I'm not sure happy with the clipboard state on linux, but this seems fine for now.

atuin 18.4.0  + ubuntu 22.04 = still no luck with copying to clipboard with ctl-y :-(

**Pull request merged**